index.js

```diff
@@ -50,51 +51,76 @@ module.exports.serverless = appFn => {
         body: template
       }
 
+      context.done()
       return
     }
 
-    // Otherwise let's listen handle the payload
-    probot = probot || loadProbot(appFn)
-
-    // Determine incoming webhook event type
-    const name = req.headers['x-github-event'] || req.headers['X-GitHub-Event']
-    const id = req.headers['x-github-delivery'] || req.headers['X-GitHub-Delivery']
-
-    // Do the thing
-    console.log(`Received event ${name}${req.body.action ? ('.' + req.body.action) : ''}`)
-
-    if (!validateSignature(req)) {
+    // Bail for null body
+    if (!req.body) {
       context.res = {
-        status: 403,
-        body: JSON.stringify({ message: 'Invalid request; signature does not match' })
+        status: 400,
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({ message: 'Event body is null' })
       }
+      context.done();
       return
     }
 
-    if (!name) {
+    // Check for correct Signature
+    if(process.env.WEBHOOK_SECRET && !isValidSignature(req)) {
       context.res = {
-        status: 400,
-        body: JSON.stringify({ message: 'Invalid request; no action' })
+        status: 403,
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({ message: 'Request signature does not match' })
       }
+      context.done();
       return
     }
 
+    // Otherwise let's listen handle the payload
+    probot = probot || loadProbot(appFn)
+
+
+    const headers = lowerCaseKeys(req.headers)
+    
+    // Determine incoming webhook event type and event ID
+    const name = headers['x-github-event']
+    const id = headers['x-github-delivery']
+    
+    // Do the thing
+    context.log(`Received event: ${name}${req.body.action ? ('.' + req.body.action) : ''}`)
+
     try {
-      await probot.receive({
-        name: name,
-        id: id,
+      await loadProbot(appFn).receive({
+        id,
+        name,
         payload: req.body
       })
```

Comment:
I think instead of the custom event verification, you should be able to do:

```js
probot.webhooks.verifyAndReceive({ id, name, payload, signature })
```

See https://github.com/octokit/webhooks.js/#webhooksverifyandreceive. Could you give that a try?

Reply:
Ahh exactly what I was after!

Reply:
is `verifyAndRecieve` the right method? Does it pass back to Probot or should I be using `probot.webhooks.verify` like:

```
//verify
probot.webhooks.verify(...)

// do the stuff
propbot.receive(...)
```

Reply:
I'm fairly sure that it will work, but didn't test. We will likely deprecate `probot.receive` for `probot.webhooks.receive`, because there will be `probot.oauth.*` APIs coming, too. I'm working on this part of Octokit/Probot right now